R/board_url.R

```diff
@@ -23,31 +23,46 @@
 #' @inheritParams new_board
 #' @export
 #' @examples
-#' github_raw <- "https://raw.githubusercontent.com/"
+#' github_raw <- function(x) paste0("https://raw.githubusercontent.com/", x)
+#'
 #' board <- board_url(c(
-#'   files = paste0(github_raw, "rstudio/pins-r/master/tests/testthat/pin-files/"),
-#'   rds = paste0(github_raw, "rstudio/pins-r/master/tests/testthat/pin-rds/"),
-#'   raw = paste0(github_raw, "rstudio/pins-r/master/tests/testthat/pin-files/first.txt")
+#'   files = github_raw("rstudio/pins-r/master/tests/testthat/pin-files/"),
+#'   rds = github_raw("rstudio/pins-r/master/tests/testthat/pin-rds/"),
+#'   raw = github_raw("rstudio/pins-r/master/tests/testthat/pin-files/first.txt")
 #' ))
 #'
 #' board %>% pin_read("rds")
 #' board %>% pin_browse("rds", local = TRUE)
 #'
 #' board %>% pin_download("files")
 #' board %>% pin_download("raw")
-board_url <- function(urls, cache = NULL, use_cache_on_failure = is_interactive()) {
-  if (!is.character(urls) || !is_named(urls)) {
-    abort("`urls` must be a named character vector")
-  }
+board_url <- function(urls,
+                      cache = NULL,
+                      use_cache_on_failure = is_interactive()) {
+
+  url_format <- get_url_format(urls)
+  versioned <-
+    switch (
+      url_format,
+      pins_yaml = {
+        manifest <- get_manifest(urls)
+        board <- board_url(manifest)
+        return(board)
+      },
+      manifest_content = TRUE,
+      vector_of_urls = FALSE,
+      board_url_error_message
```

Comment:
Yes, I don't think currently it could be possible to get to that error but I think I'd rather leave it there to error in a known way if something goes wrong in the future.